chain/consensus/filcns/upgrades.go

```diff
@@ -1995,10 +1997,25 @@ func buildUpgradeActorsV12MinerFix(oldBuggyMinerCID, newManifestCID cid.Cid) fun
 		adtStore := store.ActorStore(ctx, stateStore)
 
 		// ensure that the manifest is loaded in the blockstore
+
+		// this loads the "correct" bundle for UpgradeWatermelonFix2Height
 		if err := bundle.LoadBundles(ctx, stateStore, actorstypes.Version12); err != nil {
 			return cid.Undef, xerrors.Errorf("failed to load manifest bundle: %w", err)
 		}
 
+		// this loads the second buggy bundle, for UpgradeWatermelonFixHeight
+		_, ok := build.GetEmbeddedBuiltinActorsBundle(actorstypes.Version12, calibnetv12BuggyBundleSuffix2)
```

Comment:
So in this line, we're ensuring that we have the "second buggy bundle" as that what we will upgrade to during the first upgrade epoch ?

Reply:
Yes (but I'm missing a step), see above.